From: Olga Kornievskaia <okorniev@redhat.com>
Date: Thu, 9 Oct 2025 16:37:59 -0400
Subject: [PATCH] nfsd: add missing FATTR4_WORD2_CLONE_BLKSIZE from supported
 attributes
References: bsc#1012628
Patch-mainline: 6.17.9
Git-commit: 4d3dbc2386fe051e44efad663e0ec828b98ab53f

commit 4d3dbc2386fe051e44efad663e0ec828b98ab53f upstream.

RFC 7862 Section 4.1.2 says that if the server supports CLONE it MUST
support clone_blksize attribute.

Fixes: d6ca7d2643ee ("NFSD: Implement FATTR4_CLONE_BLKSIZE attribute")
Cc: stable@vger.kernel.org
Signed-off-by: Olga Kornievskaia <okorniev@redhat.com>
Reviewed-by: Jeff Layton <jlayton@kernel.org>
Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 fs/nfsd/nfsd.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/nfsd/nfsd.h b/fs/nfsd/nfsd.h
index ab22a4cef621..5b8e1189f0db 100644
--- a/fs/nfsd/nfsd.h
+++ b/fs/nfsd/nfsd.h
@@ -455,6 +455,7 @@ enum {
 #define NFSD4_2_SUPPORTED_ATTRS_WORD2 \
 	(NFSD4_1_SUPPORTED_ATTRS_WORD2 | \
 	FATTR4_WORD2_MODE_UMASK | \
+	FATTR4_WORD2_CLONE_BLKSIZE | \
 	NFSD4_2_SECURITY_ATTRS | \
 	FATTR4_WORD2_XATTR_SUPPORT | \
 	FATTR4_WORD2_TIME_DELEG_ACCESS | \
-- 
2.52.0

